pkg/services/apiserver/appinstaller/server.go

```diff
@@ -60,25 +60,39 @@ func (s *serverWrapper) InstallAPIGroup(apiGroupInfo *genericapiserver.APIGroupI
 				continue
 			}
 			storage := s.configureStorage(gr, dualWriteSupported, restStorage)
-			if unifiedStorage, ok := storage.(grafanarest.Storage); ok && dualWriteSupported {
-				log.Debug("Configuring dual writer for storage", "resource", gr.String(), "version", v, "storagePath", storagePath)
-				dw, err := NewDualWriter(
-					s.ctx,
-					gr,
-					s.storageOpts,
-					legacyProvider.GetLegacyStorage(gr.WithVersion(v)),
-					unifiedStorage,
-					s.kvStore,
-					s.lock,
-					s.namespaceMapper,
-					s.dualWriteService,
-					s.dualWriterMetrics,
-					s.builderMetrics,
-				)
-				if err != nil {
-					return err
+			if dualWriteSupported {
+				if unifiedStorage, ok := storage.(grafanarest.Storage); ok {
+					log.Debug("Configuring dual writer for storage", "resource", gr.String(), "version", v, "storagePath", storagePath)
+					storage, err = NewDualWriter(
+						s.ctx,
+						gr,
+						s.storageOpts,
+						legacyProvider.GetLegacyStorage(gr.WithVersion(v)),
+						unifiedStorage,
+						s.kvStore,
+						s.lock,
+						s.namespaceMapper,
+						s.dualWriteService,
+						s.dualWriterMetrics,
+						s.builderMetrics,
+					)
+					if err != nil {
+						return err
+					}
+				} else if statusRest, ok := storage.(*appsdkapiserver.StatusREST); ok {
```

Comment:
I am not proud of this nesting mess... but I think it does what we want/need and I don't think usage will be too much